kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -1327,6 +1364,117 @@ class ResidualBasedBlockBuilderAndSolver
         }
     }
 
+    /**
+     * @brief This method returns the scale norm considering for scaling the diagonal
+     * @param rA The LHS matrix
+     * @return The scale norm
+     */
+    double GetScaleNorm(TSystemMatrixType& rA)
+    {
+        if (mOptions.Is(SCALE_DIAGONAL) ) {
+            if (mScaleFactor > 1.0) {
+                return mScaleFactor;
+            } else {
+                if (mOptions.Is(CONSIDER_NORM_DIAGONAL) ) {
+                    return GetDiagonalNorm(rA);
+                } else {
+                    return TSparseSpace::TwoNorm(rA);

```

Comment:
why would u scale by the entire matrix norm?

Reply:
Because I think it makes sense (it was my original proposal if you look)